service/history/execution/mutable_state_builder.go

```diff
@@ -4707,3 +4709,41 @@ func (e *mutableStateBuilder) logDataInconsistency() {
 		tag.WorkflowRunID(runID),
 	)
 }
+
+func (e *mutableStateBuilder) closeTransactionAsMutationRemoveDeletedInfos() {
```

Comment:
I think we are fixing the symptom here; do we know why we added updates when we also wanted to delete in the first place?

Reply:
I think it's just because we didn't delete it from update info when we added delete info.
So this would gurantee that.